### DISCUSSION TITLE
course:Foundry Fundatmentals-testing part

### DISCUSSION THREAD

**Discussion by xueyuanhuang** [Cyfrin/foundry-full-course-cu#2436] [General]
Hi all,
When I learning this course, I always watching video while coding.
However, when it comes to testing part, I start to feel confusing, I don't understand why sometimes we went to `script` sometimes put files into `test`, and import different code. 

I simply just felt I don't understand all those, any suggestion for this?

**Comment by EngrPips** [comment]
Hello @xueyuanhuang, I used to be very confused about what different files are doing and why we use script or test or why sometimes we use deployment script to deploy the contract we want to test until I opened my mind and understood that taking the course is far beyond writing the code. I started making sense of everything we were doing and thinking deeply. Sometimes, I took over two days on a particular confusion, and sometimes, it took me a week to think through my confusion. My advice is is don't just write code but also think about everything you are writing and make sense of everything you write. That way, things become clear.

**Reply by xueyuanhuang** [reply]
thanks!

**Reply by EngrPips** [reply]
You are welcome.
